physicsnemo/sym/eq/spatial_grads/spatial_grads.py

```diff
@@ -18,13 +18,74 @@
 from typing import List, Optional, Union
```

Comment:
This is fine for now, but we should really upstream this neighbor-computation functionality into PhysicsNeMo core and the import it into -Sym and -CFD, rather than having separate implementations across both packages.